python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1157,12 +1157,24 @@ def build_struct_for(ctx, node, is_grouped):
                 loop_var = build_stmt(ctx, node.iter)
                 loop_indices = expr.make_var_list(size=len(loop_var.shape),
                                                   ast_builder=ctx.ast_builder)
+
                 expr_group = expr.make_expr_group(loop_indices)
                 impl.begin_frontend_struct_for(ctx.ast_builder, expr_group,
                                                loop_var)
-                ctx.create_variable(
-                    target, matrix.Vector(loop_indices,
-                                          dt=primitive_types.i32))
+                if impl.current_cfg().real_matrix:
+                    if len(loop_indices) > 1:
```

Comment:
I'm wondering if it is OK to return a vector with length 1 (as the old implementation does) so that we don't need two cases here.

Reply:
There's a special use case:
```
for I in ti.grouped(x):
    for i in range(10):
        arr[I, i, 5] = ....
```

If we return a Tensor[1 x i32] for `I`, `arr` will receiver `indices_ = [ expr(Tensor[ 1 x i32 ]), expr(i32), 5 ]` when compiling for subscription. This poses extra complicity for `_indices` expansion

Reply:
Sure, let me adjust `indices expansion` logic when compiling for subscription.